plugins/kubernetes/src/kubernetes-auth-provider/types.ts

```diff
@@ -21,6 +21,7 @@ export interface KubernetesAuthProvider {
   decorateRequestBodyForAuth(
     requestBody: KubernetesRequestBody,
   ): Promise<KubernetesRequestBody>;
+  getBearerToken(): Promise<string>;
```

Comment:
Had missed that we needed to add this. I think we should take a bit of time to make sure this is forwards compatible. At minimum I'm feeling we switch up the return value to be `Promise<{ token: string }>` so that we're able to expand it in the future. A name we use for this in many other places is `getCredentials`, with that same return value, do you think that'd be alright too?

Reply:
Clarifying question: was this suggestion just for the method getBearerToken() in the KubernetesAuthProvider interface? Or should i be renaming the getBearertoken method i introduce to the KubernetesAuthProvidersApi interface as well?

Reply:
Done